### PR TITLE
fastjet: 3.3.4 -> 3.4.0

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
&nbsp;<b style="margin: 0px; padding: 0px; caret-color: rgb(51, 51, 51); color: rgb(51, 51, 51); font-family: &quot;Gill Sans&quot;, Helvetica, Arial, sans-serif; font-size: 14px; background-color: rgb(238, 238, 238);">Main release, 25 June 2021:&nbsp;<a href="http://fastjet.fr/repo/fastjet-3.4.0.tar.gz" style="margin: 0px; padding: 0px; color: rgb(0, 110, 145);">fastjet-3.4.0</a></b><span style="caret-color: rgb(51, 51, 51); color: rgb(51, 51, 51); font-family: &quot;Gill Sans&quot;, Helvetica, Arial, sans-serif; font-size: 14px; background-color: rgb(238, 238, 238);">&nbsp;(</span><a href="http://fastjet.fr/repo/fastjet-doc-3.4.0.pdf" style="font-weight: normal; margin: 0px; padding: 0px; color: rgb(0, 110, 145); font-family: &quot;Gill Sans&quot;, Helvetica, Arial, sans-serif; font-size: 14px; background-color: rgb(238, 238, 238);">manual</a><span style="caret-color: rgb(51, 51, 51); color: rgb(51, 51, 51); font-family: &quot;Gill Sans&quot;, Helvetica, Arial, sans-serif; font-size: 14px; background-color: rgb(238, 238, 238);">,&nbsp;</span><a href="http://fastjet.fr/repo/doxygen-3.4.0" style="font-weight: normal; margin: 0px; padding: 0px; color: rgb(0, 110, 145); font-family: &quot;Gill Sans&quot;, Helvetica, Arial, sans-serif; font-size: 14px; background-color: rgb(238, 238, 238);">doxygen</a><span style="caret-color: rgb(51, 51, 51); color: rgb(51, 51, 51); font-family: &quot;Gill Sans&quot;, Helvetica, Arial, sans-serif; font-size: 14px; background-color: rgb(238, 238, 238);">,&nbsp;</span><a href="http://fastjet.fr/repo/fjcore-3.4.0.tar.gz" style="font-weight: normal; margin: 0px; padding: 0px; color: rgb(0, 110, 145); font-family: &quot;Gill Sans&quot;, Helvetica, Arial, sans-serif; font-size: 14px; background-color: rgb(238, 238, 238);">fjcore</a><span style="caret-color: rgb(51, 51, 51); color: rgb(51, 51, 51); font-family: &quot;Gill Sans&quot;, Helvetica, Arial, sans-serif; font-size: 14px; background-color: rgb(238, 238, 238);">).</span>
<ul style="margin: 0px; padding: 2px 0px 0px; caret-color: rgb(51, 51, 51); color: rgb(51, 51, 51); font-family: &quot;Gill Sans&quot;, Helvetica, Arial, sans-serif; font-size: 14px; background-color: rgb(238, 238, 238);">
    <li style="margin: 0px 0px 0px 40px; padding: 0px;">New features:
    <ul style="margin: 0px; padding: 2px 0px 0px;">
        <li style="margin: 0px 0px 0px 40px; padding: 0px;">addition of support for thread safety. It requires a compiler supporting the C++11 standard (or later) and must be explicitly enabled at configure time. There are two options
        <ul style="margin: 0px; padding: 2px 0px 0px;">
            <li style="margin: 0px 0px 0px 40px; padding: 0px;">the --enable-limited-thread-safety configure option is generally sufficient if each thread is processing entirely distinct events. It should have essentially no speed overhead.</li>
            <li style="margin: 0px 0px 0px 40px; padding: 0px;">the --enable-thread-safety configure option is required if you want to be able to read/copy/process a common set of PseudoJets across multiple threads. Depending on the system, our tests have shown speed penalties of up to 10-15%. For further details, see section 10 of the&nbsp;<a href="http://fastjet.fr/repo/fastjet-doc-3.4.0-beta.1.pdf" style="margin: 0px; padding: 0px; color: rgb(0, 110, 145);">manual</a></li>
        </ul>
        </li>
        <li style="margin: 0px 0px 0px 40px; padding: 0px;">The jet area infrastructure has new calls to facilitate reproducible multi-threaded running: AreaDefinition::with_fixed_seed(seed) and GhostedAreaSpec::get_last_seed(seed). See section 10.3.1 of the manual.</li>
        <li style="margin: 0px 0px 0px 40px; padding: 0px;">The background estimators have a new interface to facilitate reproducible multi-threaded running: BackgroundEstimatorBase::estimate(...) returns a BackgroundEstimate object, which contains rho, sigma, etc. For details on using it, see sections 8 and 10.3.2 of the manual. The Subtractor class has been updated to use this.&nbsp;</li>
    </ul>
    </li>
    <li style="margin: 0px 0px 0px 40px; padding: 0px;">Improvements to existing features:
    <ul style="margin: 0px; padding: 2px 0px 0px;">
        <li style="margin: 0px 0px 0px 40px; padding: 0px;">the Python interface now has improved exceptions (a fastjet Error is translated into a FastJetError python exception) and can be more easily be used with projects using FastJet that want to define their own python wrappers. Thanks to Patrick Komiske for the suggestions.</li>
        <li style="margin: 0px 0px 0px 40px; padding: 0px;">PxConePlugin now has an optional mode argument for e+e- running (mode=1) instead of pp running (mode=2). Thanks to Stefan Kluth for the suggestion.</li>
    </ul>
    </li>
    <li style="margin: 0px 0px 0px 40px; padding: 0px;">Bug fixes:
    <ul style="margin: 0px; padding: 2px 0px 0px;">
        <li style="margin: 0px 0px 0px 40px; padding: 0px;">the Fortran interface to e+e- interface now returns jets ordered in decreasing energy rather than decreasing transverse momentum.</li>
    </ul>
    </li>
    <li style="margin: 0px 0px 0px 40px; padding: 0px;">Changes compared to the fastjet-3.4.0-beta.1 release:
    <ul style="margin: 0px; padding: 2px 0px 0px;">
        <li style="margin: 0px 0px 0px 40px; padding: 0px;">added set_default_stream_and_mutex to Error and LimitedWarnings to help guard against garbled output.</li>
    </ul>
    </li>
</ul>

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 136359` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>fastjet</li>
    <li>fastjet-contrib</li>
    <li>herwig</li>
    <li>pythia</li>
    <li>python38Packages.rivet</li>
    <li>rivet (python39Packages.rivet)</li>
    <li>sacrifice</li>
    <li>sherpa</li>
    <li>thepeg</li>
  </ul>
</details>

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64

File: pkgs/development/libraries/physics/fastjet/default.nix [hunk 1]
```diff
@@ -2,11 +2,11 @@
...
-  version = "3.3.4";
+  version = "3.4.0";
 
   src = fetchurl {
     url = "http://fastjet.fr/repo/fastjet-${version}.tar.gz";
-    sha256 = "00zwvmnp2j79z95n9lgnq67q02bqfgirqla8j9y6jd8k3r052as3";
+    hash = "sha256-7gfIdHyOrYbYjeSp5OjR6efXYUlz9WMbqCl/egJHi5E=";
   };
 
   buildInputs = [ python2 ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/physics/fastjet/default.nix:12]: Can we switch to python3?

Comment (veprbl) [reply pkgs/development/libraries/physics/fastjet/default.nix]: Yes

**Pull request merged**